ml-proto/host/lexer.mll

```diff
@@ -90,10 +90,16 @@ let tick = '\''
 let escape = ['n''t''\\''\'''\"']
 let character = [^'"''\\''\n'] | '\\'escape | '\\'hexdigit hexdigit
 
-let num = ('+' | '-')? digit+
-let hexnum = ('+' | '-')? "0x" hexdigit+
+let sign = ('+' | '-')?
+let num = sign digit+
+let hexnum = sign "0x" hexdigit+
 let int = num | hexnum
-let float = (num '.' digit+) | num ('.' digit+)? ('e' | 'E') num
+let float = (num '.' digit+)
+          | num ('.' digit+)? ('e' | 'E') num
+          | sign "0x" hexdigit* '.'? hexdigit* 'p' sign digit*
```

Comment:
This would allow weird things like `0xp` or `0x.p` to be valid float literals. Is that intended?